cpp/src/ray/runtime/task/task_executor.cc

```diff
@@ -12,32 +12,44 @@ namespace ray {
 
 namespace internal {
 /// Execute remote functions by networking stream.
-msgpack::sbuffer TaskExecutionHandler(const std::string &func_name,
+/// If `func_ptr` is provided, execute this function directly.
+/// Otherwise, use `func_name` to lookup function pointer.
+msgpack::sbuffer TaskExecutionHandler(const void *func_ptr, const std::string &func_name,
                                       const std::vector<msgpack::sbuffer> &args_buffer,
                                       msgpack::sbuffer *actor_ptr) {
-  if (func_name.empty()) {
-    return PackError("Task function name is empty");
+  if (func_ptr == nullptr && func_name.empty()) {
+    return PackError("Both function pointer and function name of task are empty.");
   }
 
   msgpack::sbuffer result;
   do {
     try {
       if (actor_ptr) {
-        auto func_ptr = FunctionManager::Instance().GetMemberFunction(func_name);
+        const RemoteMemberFunction *fptr = nullptr;
         if (func_ptr == nullptr) {
-          result = PackError("unknown actor task: " + func_name);
-          break;
+          fptr = FunctionManager::Instance().GetMemberFunction(func_name);
```

Comment:
IIUC in `GetExecuteResult()` we look up `func_ptr` by `func_name`, then call `EntryFunction` which is this function? Are there other callsites that do not look up `func_ptr` by `func_name` before calling `TaskExecutionHandler()`? Should all callsites rely on `TaskExecutionHandler()` to look up `func_ptr` by `func_name` instead?